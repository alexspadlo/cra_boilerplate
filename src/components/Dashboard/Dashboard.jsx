import React from 'react'
import { logOut } from '../../utils'
import s from './Dashboard.module.scss'

class Dashboard extends React.Component {
    render() {
        return (
            <div className="container p-0">
                <div className="row p-5">
                    <div className="col text-center m-auto">
                        <h1 className="text-success pb-4"><strong>HEY YOU ARE IN A PRIVATE ROUTE!!!</strong></h1>
                        <button onClick={() => logOut()} className={`${s.outButton} text-center`}>Log OUT</button>
                    </div>
                </div>
            </div>

        )
    }
}

export default Dashboard
