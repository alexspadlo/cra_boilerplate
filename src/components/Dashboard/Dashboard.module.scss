.outButton {
  padding: 5px 20px;
  color: black;
  background-color: rgb(169, 184, 226);
  border: none;
  border-radius: 10px;

  &:focus {
    outline: 0;
  }

  &:hover {
    color: white;
    background-color: rgb(221, 10, 116);
  }
}
