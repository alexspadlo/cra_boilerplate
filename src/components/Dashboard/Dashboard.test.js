import React from 'react'
import { render } from '@testing-library/react'
import Dashboard from './Dashboard'

describe('Testing Dashboard Component', () => {
    test('Do not render without the props', () => {
        const { container, getByText } = render(<Dashboard />)
        expect(getByText('HEY YOU ARE IN A PRIVATE ROUTE!!!')).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot()
    })
})