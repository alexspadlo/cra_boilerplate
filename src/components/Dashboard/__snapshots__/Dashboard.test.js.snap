// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Dashboard Component Do not render without the props 1`] = `
<div
  class="container p-0"
>
  <div
    class="row p-5"
  >
    <div
      class="col text-center m-auto"
    >
      <h1
        class="text-success pb-4"
      >
        <strong>
          HEY YOU ARE IN A PRIVATE ROUTE!!!
        </strong>
      </h1>
      <button
        class="outButton text-center"
      >
        Log OUT
      </button>
    </div>
  </div>
</div>
`;
